src/ds++/test/tstSP.cc

```diff
@@ -674,7 +671,7 @@ void type_X_test(rtt_dsxx::UnitTest &ut) {
       CHECK_N_OBJECTS(3, 3, 0, 0);
 
       // assign SPfoo2 to itself
-      spfoo2 = spfoo2;
+      // spfoo2 = spfoo2; // modern compilers flag this at compile time.
```

Comment:
Does this need to stay in the code?

Reply:
I want to leave this in the code for now.  Originally, this checked that our home-grown smart-pointer class did the right thing when self-assigned (i.e. no increment of object counter).